packages/edit-site/src/components/block-editor/use-editor-iframe-props.js

```diff
@@ -55,11 +55,15 @@ export default function useEditorIframeProps() {
 				! currentPostIsTrashed
 			) {
 				event.preventDefault();
-				setCanvasMode( 'edit' );
+				history.push( { ...params, canvasMode: 'edit' }, undefined, {
```

Comment:
Are we missing `canvas: 'edit'`?

There is functionality that requires it. 

For example, the param is checked to allow navigation between editing a page's template to work, [useNavigateToPreviousEntityRecord](https://github.com/WordPress/gutenberg/blob/9ac649fda4d5f12da0c1339f371b34c3e7846621/packages/edit-site/src/components/block-editor/use-site-editor-settings.js#L29-L28).

Speaking of which, when editing a page's template I think we also need to add `canvasMode: 'edit'` to the route now:

```diff
diff --git a/packages/edit-site/src/components/block-editor/use-navigate-to-entity-record.js b/packages/edit-site/src/components/block-editor/use-navigate-to-entity-record.js
index 120b15b8551..cf44b2ae0cd 100644
--- a/packages/edit-site/src/components/block-editor/use-navigate-to-entity-record.js
+++ b/packages/edit-site/src/components/block-editor/use-navigate-to-entity-record.js
@@ -16,7 +16,9 @@ export default function useNavigateToEntityRecord() {
 
 	const onNavigateToEntityRecord = useCallback(
 		( params ) => {
-			history.push( { ...params, focusMode: true, canvas: 'edit' } );
+			history.push( { ...params, focusMode: true, canvas: 'edit', canvasMode: 'edit' }, undefined, {
+				transition: 'canvas-mode-edit-transition',
+			} );
 		},
 		[ history ]
 	);

```